python/python/res/job_queue/forward_model_status.py

```diff
@@ -0,0 +1,143 @@
+#
+#  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#  FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+#  for more details.
+import os.path
+import json
+import datetime
+import time
+import sys
+
+class ForwardModelJobStatus(object):
+
+    def __init__(self, name, start_time = None, end_time = None, status = "Waiting", error=None):
+        self.start_time = start_time
+        self.end_time = end_time
+        self.name = name
+        self.status = status
+        self.error = error
+
+
+    @classmethod
+    def load(cls,data):
+        start_time = ForwardModelStatus.deserialize_date(data["start_time"])
+        end_time = ForwardModelStatus.deserialize_date(data["end_time"])
+        name = data["name"]
+        status = data["status"]
+        error = data["error"]
+
+        return cls(name,
+                   start_time=start_time,
+                   end_time=end_time,
+                   status=status,
+                   error=error)
+
+
+    def __str__(self):
+        return "name:{} start_time:{}  end_time:{}  status:{}  error:{} ".format(self.name, self.start_time, self.end_time, self.status, self.error)
+
+
+    def dump_data(self):
+        return {"name" : self.name,
+                "status" : self.status,
+                "error" : self.error,
+                "start_time" : ForwardModelStatus.serialize_date(self.start_time),
+                "end_time" : ForwardModelStatus.serialize_date(self.end_time)}
+
+class ForwardModelStatus(object):
+    STATUS_FILE = "status.json"
+
+
+    def __init__(self, run_id, start_time = None, end_time = None):
+        self.run_id = run_id
+        self.start_time = start_time
+        self.end_time = end_time
+        self.jobs = []
+
+    @classmethod
+    def try_load(cls, status_file):
+        fp = open(status_file)
+        data = json.load(fp)
+
+        start_time = ForwardModelStatus.deserialize_date(data["start_time"])
+        end_time = ForwardModelStatus.deserialize_date(data["end_time"])
+        status = cls(data["run_id"],
+                     start_time=start_time,
+                     end_time=end_time)
+
+        for job in data["jobs"]:
+            status.add_job(ForwardModelJobStatus.load(job))
+
+        return status
+
+
+    @classmethod
+    def load(cls, path, num_retry=10):
+        attempt = 0
+        status_file = os.path.join(path, cls.STATUS_FILE)
+        while attempt < num_retry:
+            try:
+                status = cls.try_load(status_file)
+                return status
+            except:
+                time.sleep(0.10)
+                attempt += 1
+
+        return None
+
+
+    def __len__(self):
+        return len(self.jobs)
+
+
+    def __getitem__(self, index):
+        return self.jobs[index]
+
+
+    def add_job(self, job):
+        self.jobs.append(job)
```

Comment:
I think this class should make sure that what is added is indeed an instance of `ForwardModelJobStatus`.
One option could be to have an explicit assertion on the input type. 
Alternatively: 
```Python
def add_job(self, name, start_time = None, end_time = None, status = "Waiting", error=None):
    self.jobs.append( ForwardModelJobStatus(name, start_time, end_time, status, error))
```

Reply:
> I think this class should make sure that what is added is indeed an instance of ForwardModelJobStatus

I can add an `assert( isinstance( ) )` - but personally I think The Python Way&trade; is to just go along - if the duck quacks and so on???